File: qt/gen_qapplication.go
package qt

import (
	"unsafe"
)

type QApplication struct {
	h          uintptr
	isSubclass bool
}

// NewQApplication constructs a new QApplication object.
func NewQApplication(args []string) *QApplication {
	argcPtr := uintptr(unsafe.Pointer(&argc))
	var argvPtr uintptr
	if argv != nil && len(argv) > 0 {
		argvPtr = uintptr(unsafe.Pointer(&argv[0])) // 获取argv的指针
	} else {
		argvPtr = 0 // 或者使用nil
	}
	g := newQApplication(QApplication_new())
	g.isSubclass = true
	return g
}

// NewQApplication2 constructs a new QApplication object.
func NewQApplication2(args []string, param3 int) *QApplication {
	argcPtr := uintptr(unsafe.Pointer(&argc))
	var argvPtr uintptr
	if argv != nil && len(argv) > 0 {
		argvPtr = uintptr(unsafe.Pointer(&argv[0])) // 获取argv的指针
	} else {
		argvPtr = 0 // 或者使用nil
	}
	g := newQApplication(QApplication_new2((int)(param3)))
	g.isSubclass = true
	return g
}

func (this *QApplication) MetaObject() *QMetaObject {
	return newQMetaObject(QApplication_MetaObject(this.h))
}

func (this *QApplication) Metacast(param1 string) unsafe.Pointer {
	param1_Cstring := CString(param1)
	defer free(unsafe.Pointer(param1_Cstring))
	return (unsafe.Pointer)(QApplication_Metacast(this.h, param1_Cstring))
}

func QApplication_Tr(s string) string {
	s_Cstring := CString(s)
	defer free(unsafe.Pointer(s_Cstring))
	var _ms struct_miqt_string = QApplication_Tr(s_Cstring)
	_ret := GoStringN(_ms.data, int(int64(_ms.len)))
	free(unsafe.Pointer(_ms.data))
	return _ret
}

func QApplication_Style() *QStyle {
	return newQStyle(QApplication_Style())
}

func QApplication_SetStyle(style *QStyle) {
	QApplication_SetStyle(style.cPointer())
}

func QApplication_SetStyleWithStyle(style string) *QStyle {
	style_ms := struct_miqt_string{}
	style_ms.data = CString(style)
	style_ms.len = size_t(len(style))
	defer free(unsafe.Pointer(style_ms.data))
	return newQStyle(QApplication_SetStyleWithStyle(style_ms))
}

func QApplication_Palette(param1 *QWidget) *QPalette {
	_goptr := newQPalette(QApplication_Palette(param1.cPointer()))
	_goptr.GoGC() // Qt uses pass-by-value semantics for this type. Mimic with finalizer
	return _goptr
}

func QApplication_PaletteWithClassName(className string) *QPalette {
	className_Cstring := CString(className)
	defer free(unsafe.Pointer(className_Cstring))
	_goptr := newQPalette(QApplication_PaletteWithClassName(className_Cstring))
	_goptr.GoGC() // Qt uses pass-by-value semantics for this type. Mimic with finalizer
	return _goptr
}

func QApplication_SetPalette(param1 *QPalette) {
	QApplication_SetPalette(param1.cPointer())
}

func QApplication_Font() *QFont {
	_goptr := newQFont(QApplication_Font())
	_goptr.GoGC() // Qt uses pass-by-value semantics for this type. Mimic with finalizer
	return _goptr
}

func QApplication_FontWithQWidget(param1 *QWidget) *QFont {
	_goptr := newQFont(QApplication_FontWithQWidget(param1.cPointer()))
	_goptr.GoGC() // Qt uses pass-by-value semantics for this type. Mimic with finalizer
	return _goptr
}

func QApplication_FontWithClassName(className string) *QFont {
	className_Cstring := CString(className)
	defer free(unsafe.Pointer(className_Cstring))
	_goptr := newQFont(QApplication_FontWithClassName(className_Cstring))
	_goptr.GoGC() // Qt uses pass-by-value semantics for this type. Mimic with finalizer
	return _goptr
}

func QApplication_SetFont(param1 *QFont) {
	QApplication_SetFont(param1.cPointer())
}

func QApplication_FontMetrics() *QFontMetrics {
	_goptr := newQFontMetrics(QApplication_FontMetrics())
	_goptr.GoGC() // Qt uses pass-by-value semantics for this type. Mimic with finalizer
	return _goptr
}

func QApplication_AllWidgets() []*QWidget {
	var _ma struct_miqt_array = QApplication_AllWidgets()
	_ret := make([]*QWidget, int(_ma.len))
	_outCast := (*[0xffff]*QWidget)(unsafe.Pointer(_ma.data)) // hey ya
	for i := 0; i < int(_ma.len); i++ {
		_ret[i] = newQWidget(_outCast[i])
	}
	return _ret
}

func QApplication_TopLevelWidgets() []*QWidget {
	var _ma struct_miqt_array = QApplication_TopLevelWidgets()
	_ret := make([]*QWidget, int(_ma.len))
	_outCast := (*[0xffff]*QWidget)(unsafe.Pointer(_ma.data)) // hey ya
	for i := 0; i < int(_ma.len); i++ {
		_ret[i] = newQWidget(_outCast[i])
	}
	return _ret
}

func QApplication_ActivePopupWidget() *QWidget {
	return newQWidget(QApplication_ActivePopupWidget())
}

func QApplication_ActiveModalWidget() *QWidget {
	return newQWidget(QApplication_ActiveModalWidget())
}

func QApplication_FocusWidget() *QWidget {
	return newQWidget(QApplication_FocusWidget())
}

func QApplication_ActiveWindow() *QWidget {
	return newQWidget(QApplication_ActiveWindow())
}

func QApplication_SetActiveWindow(act *QWidget) {
	QApplication_SetActiveWindow(act.cPointer())
}

func QApplication_WidgetAt(p *QPoint) *QWidget {
	return newQWidget(QApplication_WidgetAt(p.cPointer()))
}

func QApplication_WidgetAt2(x int, y int) *QWidget {
	return newQWidget(QApplication_WidgetAt2((int)(x), (int)(y)))
}

func QApplication_TopLevelAt(p *QPoint) *QWidget {
	return newQWidget(QApplication_TopLevelAt(p.cPointer()))
}

func QApplication_TopLevelAt2(x int, y int) *QWidget {
	return newQWidget(QApplication_TopLevelAt2((int)(x), (int)(y)))
}

func QApplication_Beep() {
	QApplication_Beep()
}

func QApplication_Alert(widget *QWidget) {
	QApplication_Alert(widget.cPointer())
}

func QApplication_SetCursorFlashTime(cursorFlashTime int) {
	QApplication_SetCursorFlashTime((int)(cursorFlashTime))
}

func QApplication_CursorFlashTime() int {
	return (int)(QApplication_CursorFlashTime())
}

func QApplication_SetDoubleClickInterval(doubleClickInterval int) {
	QApplication_SetDoubleClickInterval((int)(doubleClickInterval))
}

func QApplication_DoubleClickInterval() int {
	return (int)(QApplication_DoubleClickInterval())
}

func QApplication_SetKeyboardInputInterval(keyboardInputInterval int) {
	QApplication_SetKeyboardInputInterval((int)(keyboardInputInterval))
}

func QApplication_KeyboardInputInterval() int {
	return (int)(QApplication_KeyboardInputInterval())
}

func QApplication_SetWheelScrollLines(wheelScrollLines int) {
	QApplication_SetWheelScrollLines((int)(wheelScrollLines))
}

func QApplication_WheelScrollLines() int {
	return (int)(QApplication_WheelScrollLines())
}

func QApplication_SetStartDragTime(ms int) {
	QApplication_SetStartDragTime((int)(ms))
}

func QApplication_StartDragTime() int {
	return (int)(QApplication_StartDragTime())
}

func QApplication_SetStartDragDistance(l int) {
	QApplication_SetStartDragDistance((int)(l))
}

func QApplication_StartDragDistance() int {
	return (int)(QApplication_StartDragDistance())
}

func QApplication_IsEffectEnabled(param1 UIEffect) bool {
	return (bool)(QApplication_IsEffectEnabled((int)(param1)))
}

func QApplication_SetEffectEnabled(param1 UIEffect) {
	QApplication_SetEffectEnabled((int)(param1))
}

func QApplication_Exec() int {
	return (int)(QApplication_Exec())
}

func (this *QApplication) Notify(param1 *QObject, param2 *QEvent) bool {
	return (bool)(QApplication_Notify(this.h, param1.cPointer(), param2.cPointer()))
}

func (this *QApplication) FocusChanged(old *QWidget, now *QWidget) {
	QApplication_FocusChanged(this.h, old.cPointer(), now.cPointer())
}

func (this *QApplication) OnFocusChanged(slot func(old *QWidget, now *QWidget)) {
	QApplication_connect_FocusChanged(this.h, intptr_t(cgo.NewHandle(slot)))
}

//export miqt_exec_callback_QApplication_FocusChanged
func miqt_exec_callback_QApplication_FocusChanged(cb intptr_t, old *QWidget, now *QWidget) {
	gofunc, ok := cgo.Handle(cb).Value().(func(old *QWidget, now *QWidget))
	if !ok {
		panic("miqt: callback of non-callback type (heap corruption?)")
	}

	// Convert all CABI parameters to Go parameters
	slotval1 := newQWidget(old)

	slotval2 := newQWidget(now)

	gofunc(slotval1, slotval2)
}

func (this *QApplication) StyleSheet() string {
	var _ms struct_miqt_string = QApplication_StyleSheet(this.h)
	_ret := GoStringN(_ms.data, int(int64(_ms.len)))
	free(unsafe.Pointer(_ms.data))
	return _ret
}

func (this *QApplication) AutoSipEnabled() bool {
	return (bool)(QApplication_AutoSipEnabled(this.h))
}

func (this *QApplication) SetStyleSheet(sheet string) {
	sheet_ms := struct_miqt_string{}
	sheet_ms.data = CString(sheet)
	sheet_ms.len = size_t(len(sheet))
	defer free(unsafe.Pointer(sheet_ms.data))
	QApplication_SetStyleSheet(this.h, sheet_ms)
}

func (this *QApplication) SetAutoSipEnabled(enabled bool) {
	QApplication_SetAutoSipEnabled(this.h, (bool)(enabled))
}

func QApplication_CloseAllWindows() {
	QApplication_CloseAllWindows()
}

func QApplication_AboutQt() {
	QApplication_AboutQt()
}

func QApplication_Tr2(s string, c string) string {
	s_Cstring := CString(s)
	defer free(unsafe.Pointer(s_Cstring))
	c_Cstring := CString(c)
	defer free(unsafe.Pointer(c_Cstring))
	var _ms struct_miqt_string = QApplication_Tr2(s_Cstring, c_Cstring)
	_ret := GoStringN(_ms.data, int(int64(_ms.len)))
	free(unsafe.Pointer(_ms.data))
	return _ret
}

func QApplication_Tr3(s string, c string, n int) string {
	s_Cstring := CString(s)
	defer free(unsafe.Pointer(s_Cstring))
	c_Cstring := CString(c)
	defer free(unsafe.Pointer(c_Cstring))
	var _ms struct_miqt_string = QApplication_Tr3(s_Cstring, c_Cstring, (int)(n))
	_ret := GoStringN(_ms.data, int(int64(_ms.len)))
	free(unsafe.Pointer(_ms.data))
	return _ret
}

func QApplication_SetPalette2(param1 *QPalette, className string) {
	className_Cstring := CString(className)
	defer free(unsafe.Pointer(className_Cstring))
	QApplication_SetPalette2(param1.cPointer(), className_Cstring)
}

func QApplication_SetFont2(param1 *QFont, className string) {
	className_Cstring := CString(className)
	defer free(unsafe.Pointer(className_Cstring))
	QApplication_SetFont2(param1.cPointer(), className_Cstring)
}

func QApplication_Alert2(widget *QWidget, duration int) {
	QApplication_Alert2(widget.cPointer(), (int)(duration))
}

func QApplication_SetEffectEnabled2(param1 UIEffect, enable bool) {
	QApplication_SetEffectEnabled2((int)(param1), (bool)(enable))
}

func (this *QApplication) callVirtualBase_MetaObject() *QMetaObject {
	return newQMetaObject(QApplication_virtualbase_MetaObject(unsafe.Pointer(this.h)))
}

func (this *QApplication) OnMetaObject(slot func(super func() *QMetaObject) *QMetaObject) {
	if !this.isSubclass {
		panic("miqt: can only override virtual methods for directly constructed types")
	}
	QApplication_override_virtual_MetaObject(unsafe.Pointer(this.h), intptr_t(cgo.NewHandle(slot)))
}

//export miqt_exec_callback_QApplication_MetaObject
func miqt_exec_callback_QApplication_MetaObject(self QApplication, cb intptr_t) *QMetaObject {
	gofunc, ok := cgo.Handle(cb).Value().(func(super func() *QMetaObject) *QMetaObject)
	if !ok {
		panic("miqt: callback of non-callback type (heap corruption?)")
	}

	virtualReturn := gofunc((&QApplication{h: self}).callVirtualBase_MetaObject)

	return virtualReturn.cPointer()
}

func (this *QApplication) callVirtualBase_Metacast(param1 string) unsafe.Pointer {
	param1_Cstring := CString(param1)
	defer free(unsafe.Pointer(param1_Cstring))

	return (unsafe.Pointer)(QApplication_virtualbase_Metacast(unsafe.Pointer(this.h), param1_Cstring))
}

func (this *QApplication) OnMetacast(slot func(super func(param1 string) unsafe.Pointer, param1 string) unsafe.Pointer) {
	if !this.isSubclass {
		panic("miqt: can only override virtual methods for directly constructed types")
	}
	QApplication_override_virtual_Metacast(unsafe.Pointer(this.h), intptr_t(cgo.NewHandle(slot)))
}

//export miqt_exec_callback_QApplication_Metacast
func miqt_exec_callback_QApplication_Metacast(self QApplication, cb intptr_t, param1 *const_char) unsafe.Pointer {
	gofunc, ok := cgo.Handle(cb).Value().(func(super func(param1 string) unsafe.Pointer, param1 string) unsafe.Pointer)
	if !ok {
		panic("miqt: callback of non-callback type (heap corruption?)")
	}
	// Convert all CABI parameters to Go parameters
	param1_ret := param1
	slotval1 := GoString(param1_ret)
	virtualReturn := gofunc((&QApplication{h: self}).callVirtualBase_Metacast, slotval1)

	return virtualReturn
}

File: cmd/genbindings/cabiTest/gen_qapplication.cpp
// +build ignore

#include <QApplication>
#include <QCoreApplication>
#include <QEvent>
#include <QFont>
#include <QFontMetrics>
#include <QGuiApplication>
#include <QList>
#include <QMetaObject>
#include <QObject>
#include <QPalette>
#include <QPoint>
#include <QString>
#include <QByteArray>
#include <cstring>
#include <QStyle>
#include <QWidget>
#include <qapplication.h>
#include "gen_qapplication.h"

#ifndef _Bool
#define _Bool bool
#endif

void _GUID_Delete(_GUID* self, bool isSubclass) {
    delete self;
}

void type_info_Delete(type_info* self, bool isSubclass) {
	if (isSubclass) {
		delete dynamic_cast<type_info*>( self );
	} else {
		delete self;
	}
}
typedef bool (*CallbackFunc)( int argc, void* args);

bool miqt_exec_callback_QApplication_Notify(  CallbackFunc cb, int argc, void* args) {
    if (cb == NULL) {
        fprintf(stderr, "miqt: callback is NULL (heap corruption?)\n");
        return false; // 如果回调函数为空，返回错误
    }
    return cb(argc, args);
}


QApplication* QApplication_new(int* argc, char** argv) {
	return new QApplication ( *argc, argv);
}

QApplication* QApplication_new2(int* argc, char** argv, int param3) {
	return new QApplication(static_cast<int&>(*argc), argv, static_cast<int>(param3));
}

void QApplication_virtbase(QApplication* src, QGuiApplication** outptr_QGuiApplication) {
	*outptr_QGuiApplication = static_cast<QGuiApplication*>(src);
}

QMetaObject* QApplication_MetaObject(const QApplication* self) {
	return (QMetaObject*) self->metaObject();
}

void* QApplication_Metacast(QApplication* self, const char* param1) {
	return self->qt_metacast(param1);
}

struct miqt_string QApplication_Tr(const char* s) {
	QString _ret = QApplication::tr(s);
	// Convert QString from UTF-16 in C++ RAII memory to UTF-8 in manually-managed C memory
	QByteArray _b = _ret.toUtf8();
	struct miqt_string _ms;
	_ms.len = _b.length();
	_ms.data = static_cast<char*>(malloc(_ms.len));
	memcpy(_ms.data, _b.data(), _ms.len);
	return _ms;
}

QStyle* QApplication_Style() {
	return QApplication::style();
}

void QApplication_SetStyle(QStyle* style) {
	QApplication::setStyle(style);
}

QStyle* QApplication_SetStyleWithStyle(struct miqt_string style) {
	QString style_QString = QString::fromUtf8(style.data, style.len);
	return QApplication::setStyle(style_QString);
}

QPalette* QApplication_Palette(QWidget* param1) {
	return new QPalette(QApplication::palette(param1));
}

QPalette* QApplication_PaletteWithClassName(const char* className) {
	return new QPalette(QApplication::palette(className));
}

void QApplication_SetPalette(QPalette* param1) {
	QApplication::setPalette(*param1);
}

QFont* QApplication_Font() {
	return new QFont(QApplication::font());
}

QFont* QApplication_FontWithQWidget(QWidget* param1) {
	return new QFont(QApplication::font(param1));
}

QFont* QApplication_FontWithClassName(const char* className) {
	return new QFont(QApplication::font(className));
}

void QApplication_SetFont(QFont* param1) {
	QApplication::setFont(*param1);
}

QFontMetrics* QApplication_FontMetrics() {
	return new QFontMetrics(QApplication::fontMetrics());
}

struct miqt_array /* of QWidget* */  QApplication_AllWidgets() {
	QWidgetList _ret = QApplication::allWidgets();
	// Convert QList<> from C++ memory to manually-managed C memory
	QWidget** _arr = static_cast<QWidget**>(malloc(sizeof(QWidget*) * _ret.length()));
	for (size_t i = 0, e = _ret.length(); i < e; ++i) {
		_arr[i] = _ret[i];
	}
	struct miqt_array _out;
	_out.len = _ret.length();
	_out.data = static_cast<void*>(_arr);
	return _out;
}

struct miqt_array /* of QWidget* */  QApplication_TopLevelWidgets() {
	QWidgetList _ret = QApplication::topLevelWidgets();
	// Convert QList<> from C++ memory to manually-managed C memory
	QWidget** _arr = static_cast<QWidget**>(malloc(sizeof(QWidget*) * _ret.length()));
	for (size_t i = 0, e = _ret.length(); i < e; ++i) {
		_arr[i] = _ret[i];
	}
	struct miqt_array _out;
	_out.len = _ret.length();
	_out.data = static_cast<void*>(_arr);
	return _out;
}

QWidget* QApplication_ActivePopupWidget() {
	return QApplication::activePopupWidget();
}

QWidget* QApplication_ActiveModalWidget() {
	return QApplication::activeModalWidget();
}

QWidget* QApplication_FocusWidget() {
	return QApplication::focusWidget();
}

QWidget* QApplication_ActiveWindow() {
	return QApplication::activeWindow();
}

void QApplication_SetActiveWindow(QWidget* act) {
	QApplication::setActiveWindow(act);
}

QWidget* QApplication_WidgetAt(QPoint* p) {
	return QApplication::widgetAt(*p);
}

QWidget* QApplication_WidgetAt2(int x, int y) {
	return QApplication::widgetAt(static_cast<int>(x), static_cast<int>(y));
}

QWidget* QApplication_TopLevelAt(QPoint* p) {
	return QApplication::topLevelAt(*p);
}

QWidget* QApplication_TopLevelAt2(int x, int y) {
	return QApplication::topLevelAt(static_cast<int>(x), static_cast<int>(y));
}

void QApplication_Beep() {
	QApplication::beep();
}

void QApplication_Alert(QWidget* widget) {
	QApplication::alert(widget);
}

void QApplication_SetCursorFlashTime(int cursorFlashTime) {
	QApplication::setCursorFlashTime(static_cast<int>(cursorFlashTime));
}

int QApplication_CursorFlashTime() {
	return QApplication::cursorFlashTime();
}

void QApplication_SetDoubleClickInterval(int doubleClickInterval) {
	QApplication::setDoubleClickInterval(static_cast<int>(doubleClickInterval));
}

int QApplication_DoubleClickInterval() {
	return QApplication::doubleClickInterval();
}

void QApplication_SetKeyboardInputInterval(int keyboardInputInterval) {
	QApplication::setKeyboardInputInterval(static_cast<int>(keyboardInputInterval));
}

int QApplication_KeyboardInputInterval() {
	return QApplication::keyboardInputInterval();
}

void QApplication_SetWheelScrollLines(int wheelScrollLines) {
	QApplication::setWheelScrollLines(static_cast<int>(wheelScrollLines));
}

int QApplication_WheelScrollLines() {
	return QApplication::wheelScrollLines();
}

void QApplication_SetStartDragTime(int ms) {
	QApplication::setStartDragTime(static_cast<int>(ms));
}

int QApplication_StartDragTime() {
	return QApplication::startDragTime();
}

void QApplication_SetStartDragDistance(int l) {
	QApplication::setStartDragDistance(static_cast<int>(l));
}

int QApplication_StartDragDistance() {
	return QApplication::startDragDistance();
}

bool QApplication_IsEffectEnabled(int param1) {
	return QApplication::isEffectEnabled(static_cast<Qt::UIEffect>(param1));
}

void QApplication_SetEffectEnabled(int param1) {
	QApplication::setEffectEnabled(static_cast<Qt::UIEffect>(param1));
}

int QApplication_Exec(QApplication *self) {
    return static_cast<QApplication*>(self)->exec();
}

bool QApplication_Notify(QApplication* self, QObject* param1, QEvent* param2) {
	return self->notify(param1, param2);
}

void QApplication_FocusChanged(QApplication* self, QWidget* old, QWidget* now) {
	self->focusChanged(old, now);
}

void QApplication_connect_FocusChanged(QApplication* self, intptr_t slot) {
	QApplication::connect(self, static_cast<void (QApplication::*)(QWidget*, QWidget*)>(&QApplication::focusChanged), self, [=](QWidget* old, QWidget* now) {
		QWidget* sigval1 = old;
		QWidget* sigval2 = now;
        //todo
//		miqt_exec_callback_QApplication_FocusChanged(slot, sigval1, sigval2);
//        miqt_exec_callback_QApplication_Notify(slot, sigval1, sigval2);
	});
}

struct miqt_string QApplication_StyleSheet(const QApplication* self) {
	QString _ret = self->styleSheet();
	// Convert QString from UTF-16 in C++ RAII memory to UTF-8 in manually-managed C memory
	QByteArray _b = _ret.toUtf8();
	struct miqt_string _ms;
	_ms.len = _b.length();
	_ms.data = static_cast<char*>(malloc(_ms.len));
	memcpy(_ms.data, _b.data(), _ms.len);
	return _ms;
}

bool QApplication_AutoSipEnabled(const QApplication* self) {
	return self->autoSipEnabled();
}

void QApplication_SetStyleSheet(QApplication* self, struct miqt_string sheet) {
	QString sheet_QString = QString::fromUtf8(sheet.data, sheet.len);
	self->setStyleSheet(sheet_QString);
}

void QApplication_SetAutoSipEnabled(QApplication* self, const bool enabled) {
	self->setAutoSipEnabled(enabled);
}

void QApplication_CloseAllWindows() {
	QApplication::closeAllWindows();
}

void QApplication_AboutQt() {
	QApplication::aboutQt();
}

struct miqt_string QApplication_Tr2(const char* s, const char* c) {
	QString _ret = QApplication::tr(s, c);
	// Convert QString from UTF-16 in C++ RAII memory to UTF-8 in manually-managed C memory
	QByteArray _b = _ret.toUtf8();
	struct miqt_string _ms;
	_ms.len = _b.length();
	_ms.data = static_cast<char*>(malloc(_ms.len));
	memcpy(_ms.data, _b.data(), _ms.len);
	return _ms;
}

struct miqt_string QApplication_Tr3(const char* s, const char* c, int n) {
	QString _ret = QApplication::tr(s, c, static_cast<int>(n));
	// Convert QString from UTF-16 in C++ RAII memory to UTF-8 in manually-managed C memory
	QByteArray _b = _ret.toUtf8();
	struct miqt_string _ms;
	_ms.len = _b.length();
	_ms.data = static_cast<char*>(malloc(_ms.len));
	memcpy(_ms.data, _b.data(), _ms.len);
	return _ms;
}

void QApplication_SetPalette2(QPalette* param1, const char* className) {
	QApplication::setPalette(*param1, className);
}

void QApplication_SetFont2(QFont* param1, const char* className) {
	QApplication::setFont(*param1, className);
}

void QApplication_Alert2(QWidget* widget, int duration) {
	QApplication::alert(widget, static_cast<int>(duration));
}

void QApplication_SetEffectEnabled2(int param1, bool enable) {
	QApplication::setEffectEnabled(static_cast<Qt::UIEffect>(param1), enable);
}

void QApplication_override_virtual_Notify(void* self, intptr_t slot) {
//	dynamic_cast<QApplication*>( (QApplication*)(self) )->handle__Notify = slot;
}

bool QApplication_virtualbase_Notify(void* self, QObject* param1, QEvent* param2) {
//	return ( (QApplication*)(self) )->virtualbase_Notify(param1, param2);
    return false;
}

void QApplication_override_virtual_Event(void* self, intptr_t slot) {
//	dynamic_cast<QApplication*>( (QApplication*)(self) )->handle__Event = slot;
}

bool QApplication_virtualbase_Event(void* self, QEvent* param1) {
//	return ( (QApplication*)(self) )->virtualbase_Event(param1);
    return false;
}

void QApplication_Delete(QApplication* self, bool isSubclass) {
	if (isSubclass) {
		delete dynamic_cast<QApplication*>( self );
	} else {
		delete self;
	}
}

